packages/webapp/pages/cores/index.tsx

```diff
@@ -164,6 +167,18 @@ export const CorePageRenderer = ({
     paddle,
   ]);
 
+  useEffect(() => {
+    if (checkCoresRoleNotNone(user)) {
+      return;
+    }
+
+    router.push(webappUrl);
+  }, [router, user]);
+
+  if (!router.isReady) {
+    return null;
```

Comment:
Also return null when user should not see so that content does not flash for a brief second if JS is slower to load

Reply:
Added